front/src/modules/utils/cookie-storage.ts

```diff
@@ -0,0 +1,62 @@
+import Cookies, { CookieAttributes } from 'js-cookie';
+
+type Listener = (
+  newValue: string | undefined,
+  oldValue: string | undefined,
+) => void;
+
+class CookieStorage {
+  private listeners: Record<string, Listener[]> = {};
+  private keys: Set<string> = new Set();
+
+  getItem(key: string): string | undefined {
+    return Cookies.get(key);
+  }
+
+  setItem(key: string, value: string, attributes?: CookieAttributes): void {
+    const oldValue = this.getItem(key);
+
+    this.keys.add(key);
+    Cookies.set(key, value, attributes);
+    this.dispatch(key, value, oldValue);
+  }
+
+  removeItem(key: string): void {
+    const oldValue = this.getItem(key);
+
+    this.keys.delete(key);
+    Cookies.remove(key);
+    this.dispatch(key, undefined, oldValue);
+  }
+
+  clear(): void {
+    this.keys.forEach((key) => this.removeItem(key));
+  }
+
+  private dispatch(
```

Comment:
I would try to rely on Recoil for handling state across components as it seems to be feasible and this implementation would introduce a custom event handling system in parallel.

Reply:
I will take a look on Recoil to see how I can change this